Modules/_decimal/docstrings.h

```diff
@@ -17,24 +17,154 @@
 
 
 PyDoc_STRVAR(doc__decimal,
-"C decimal arithmetic module");
+"This is an implementation of decimal floating point arithmetic based on\n\
```

Comment:
Instead of copying the module documentation in `_decimal` module, why not simply move it in `decimal.py`? You can keep a simpler docstring in `_pydecimal.py`

Reply:
> Instead of copying the module documentation in `_decimal` module, why not simply move it in `decimal.py`? `

```python
$ cat ./Lib/decimal.py   

try:
    from _decimal import *
    from _decimal import __doc__
    from _decimal import __version__
    from _decimal import __libmpdec_version__
except ImportError:
    from _pydecimal import *
    from _pydecimal import __doc__
    from _pydecimal import __version__
    from _pydecimal import __libmpdec_version__
```
I'm not sure what you mean here; is it going to be a dynamic structure, as Raymond Hattinger [mentioned](https://github.com/python/cpython/issues/71966#issuecomment-1093722871)?

```python
for name in dir(_decimal.Decimal):
    if name.startswith('_'):
        continue
    py_method = getattr(_decimal.Decimal, name)
    py_doc = py_method.__doc__
    if py_doc is None:
        continue
    c_method = getattr(_pydecimal.Decimal, name, None)
    if c_method is None:
        continue
    c_doc = c_method.__doc__
    if c_doc is None or len(c_doc) < len(py_doc):
        c_method.__doc__ = py_doc
 ```

Reply:
Remove `from ... import __doc__` and move the module docstring from `_pydecimal` to `decimal`. Only left "Python decimal arithmetic module" in `_pydecimal`.

Reply:
My comment was only about the module docstring.

Reply:
> You can keep a simpler docstring in `_pydecimal.py`

Actually, when I first looked at the issue, there were two patches. And these patches did not only include synchronization.

For example, for `quantize($self, a, b, /),`  _The coefficient of the result is derived from that of the left-hand operand._ was added from [this](https://speleotrove.com/decimal/daops.html#refquant) source.

Therefore, I tried to add the new changes in the patch to the `_pydecimal.py` and `docstrings.h` files.

Reply:
I was only taking about the module docstring here.

The newer patch is the updated version of the earlier patch.

Dynamic update of docstrings in the C implementation looks interesting, but does not work, because they are read-only. And even if they would not, we should consider performance impact which can be noticeable.

Reply:
If it was unclear what I proposed, I created a separate PR #117919 for this change.